hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -97,28 +108,69 @@ def get_neighbors(array, i, j, n=1):
     return array[i_start:i_stop, j_start:j_stop]
 
 
-def ref_point_with_max_cc(data_cc: np.array, window_size=1):
-    '''
-    shift determine the window size, n=1 9-pixel window, n=2, 25-pixel window, etc.
-    '''
-
-    data_cc_max = data_cc[data_cc < 1.0].max()
-    idx = np.where(data_cc == data_cc_max)
+def find_ref_point_from_candidates(data_cc: np.ndarray, indices: (np.array, np.array),
+                                   window_size, start_idx, pick_num):
+    i_max = indices[0].size
+    rows = indices[0][start_idx:min(start_idx + pick_num, i_max)]
+    cols = indices[1][start_idx:min(start_idx + pick_num, i_max)]
 
-    rows, cols = idx[0], idx[1]
     num = len(rows)
     tots = np.zeros(num, dtype=float)
 
     for k in range(num):
-        tots[k] = get_neighbors(data_cc, rows[k], cols[k], window_size).sum()
-
-    idx = np.where(tots == tots.max())
-    ref_i = rows[idx[0][0]]
-    ref_j = cols[idx[0][0]]
+        neighbors = get_neighbors(data_cc, rows[k], cols[k], window_size)
+        if not ((neighbors == 0.0).any() or neighbors.size < (2*window_size+1)**2):
+            tots[k] = neighbors.sum()
+
+    if (tots == 0.0).all():
+        ref_i = None
+        ref_j = None
+    else:
+        idx = np.where(tots == tots.max())
+        ref_i = rows[idx[0][0]]
+        ref_j = cols[idx[0][0]]
 
     return ref_i, ref_j
 
 
+def ref_point_with_max_cc(data_cc: np.array, window_size=10, pick_num=20, cc_thresh=0.3):
+    """
+    Args:
+        data_cc: array includes cc data
+        window_size: the actual window is a square of size (2*window_size + 1);
+        for example, window_size=1 means 9-pixel window, window_size=2 means 25-pixel window, etc.
+        pick_num: the number of elements
+        cc_thresh: cc threshold used to exclude the pixels seeking the reference pixel
+
+    Returns:
+        indices of the reference pixel
+    """
+
+    data = data_cc.copy()
+
+    data[np.logical_or(data == 1.0, data < cc_thresh)] = 0.0
+
+    indices = np.unravel_index(np.argsort(-data_cc, axis=None), data_cc.shape)
+
+    start_idx = 0
+
+    num = data_cc.size
+
+    while start_idx < num:
```

Comment:
We can simplify this:
```suggestion
    while start_idx < data_cc.size:
```

Reply:
I am thinking put data_cc.size in the while loop means that each loop, it will run data_cc.size, which waste the computing time. If it is a variable in the while loop, it does not calculate each loop, it will be faster?

Reply:
No, `size` is just an attribute of the `data_cc` object, so it is already computed.